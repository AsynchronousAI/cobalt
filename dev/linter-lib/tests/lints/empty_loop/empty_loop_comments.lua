for _ = 1, 10 do
    return "Should not warn"
end

for _ = 1, 10 do
    print("Should not warn")
end

for _ = 1, 10 do
    -- Should not warn
end

for _ = 1, 10 do
    --[[
    Should not warn
    ]]
end

for _ = 1, 10 do



end

for _ in pairs({}) do
    return "Should not warn"
end

for _ in pairs({}) do
end

for _ in ipairs({}) do
    return "Should not warn"
end

for _ in ipairs({}) do
end

for _ in {} do
    return "Should not warn"
end

for _ in {} do
end

for _ in a() do
    return "Should not warn"
end

for _ in a() do
end

while true do
    return "Should not warn"
end

while true do
end

repeat
    return "Should not warn"
until true

repeat
until true
-- comment here shouldn't break anything
