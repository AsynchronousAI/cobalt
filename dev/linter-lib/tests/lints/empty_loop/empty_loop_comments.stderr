error[empty_loop]: empty loop block
   ┌─ empty_loop_comments.lua:19:1
   │  
19 │ ╭ for _ = 1, 10 do
20 │ │ 
21 │ │ 
22 │ │ 
23 │ │ end
   │ ╰───^

error[empty_loop]: empty loop block
   ┌─ empty_loop_comments.lua:29:1
   │  
29 │ ╭ for _ in pairs({}) do
30 │ │ end
   │ ╰───^

error[empty_loop]: empty loop block
   ┌─ empty_loop_comments.lua:36:1
   │  
36 │ ╭ for _ in ipairs({}) do
37 │ │ end
   │ ╰───^

error[empty_loop]: empty loop block
   ┌─ empty_loop_comments.lua:43:1
   │  
43 │ ╭ for _ in {} do
44 │ │ end
   │ ╰───^

error[empty_loop]: empty loop block
   ┌─ empty_loop_comments.lua:50:1
   │  
50 │ ╭ for _ in a() do
51 │ │ end
   │ ╰───^

error[empty_loop]: empty loop block
   ┌─ empty_loop_comments.lua:57:1
   │  
57 │ ╭ while true do
58 │ │ end
   │ ╰───^

error[empty_loop]: empty loop block
   ┌─ empty_loop_comments.lua:64:1
   │  
64 │ ╭ repeat
65 │ │ until true
   │ ╰──────────^

